gpflow/utilities/ops.py

```diff
@@ -2,9 +2,17 @@
 from typing import List, Optional, Union
 
 import tensorflow as tf
-import tensorflow_probability as tfp
 import numpy as np
-from ..base import cast
+
+
+def cast(
```

Comment:
@awav what makes this code duplication better than importing it from base ?

Reply:
The `base` should not export `cast`, this is simply a very bad place for a utility function.
The `tf.cast` is TF issue, we should not break gpflow integrity because of TF mistakes.

Reply:
I'm still trying to understand here... if it's used all over the place, why is `base` a bad module for it ?

Reply:
> if it's used all over the place, why is base a bad module for it?

The cast is a utility function and proper place is the utility module. Making it public though base module is wrong in terms of package hierarchy.

Because it is a temporary issue, I could make a cast function inside the base and use it in utilities, but export only utility function. This would help to avoid duplication. 

@st-- do you want me to make this change?

Reply:
No it's fine - hopefully it'll be fixed upstream soon and we can remove all of it anyways! - I was just trying to understand your reasoning :)